libkbfs/folder_branch_ops.go

```diff
@@ -4949,7 +4953,9 @@ func (fbo *folderBranchOps) registerAndWaitForUpdates() {
 		// Never give up hope until we shut down
 		expBackoff.MaxElapsedTime = 0
 		// Register and wait in a loop unless we hit an unrecoverable error
-		ctx, cancel := context.WithCancel(ctx)
+		fbo.cancelUpdatesLock.Lock()
+		ctx, fbo.cancelUpdates = context.WithCancel(ctx)
```

Comment:
Since we're enforcing there can only be one update goroutine at a time, we can safely overwrite any existing `cancelUpdates` here.  Whatever the old one was, it's useless now.